Core/clim-basic/utils.lisp

```diff
@@ -743,3 +728,195 @@ a flag CLOSED is T then beginning and end of the list are consecutive too."
+                                       current-margin)))))))
+    (break-line)))
+
+;;; Break greedily on the break opportunity. If a single word doesn't fit in the
+;;; line and that line is shorter than an empty line then we break it right
+;;; away. If the line is longer or equal to an empty one then we break this word
+;;; by a character. If there are multiple #\space characters at the line
+;;; beginning they are kept in the previous line.
+(defun %line-breaks-2 (string width initial-offset margin start end opportunities
+                       &aux (width-fn (etypecase width
+                                        (function width)
+                                        (number (lambda (string start end)
+                                                  (declare (ignore string))
+                                                  (* width (- end start)))))))
+  (labels ((skip-whitespace (string index)
+             (and index (position #\space string :start index :test-not #'char=)))
+           (breaks-rec (offset margin start end next-opportunity-index
+                        &aux (current-margin (max (- margin offset) 1)))
+             (cond ((<= (funcall width-fn string start end) current-margin)
+                    (return-from breaks-rec
+                      nil))
+                   ((= next-opportunity-index (length opportunities))
+                    (return-from breaks-rec
+                      (%line-breaks-1 string width offset margin start end)))
+                   ((>= start (aref opportunities next-opportunity-index))
+                    (return-from breaks-rec
+                      (breaks-rec offset margin start end (1+ next-opportunity-index)))))
+             (loop
+                with best-break = nil
+                for i from next-opportunity-index below (length opportunities)
+                for opportunity = (aref opportunities i)
+                for current-width = (funcall width-fn string start opportunity)
+                do
+                  (cond ((<= current-width current-margin)
+                         (if-let ((pos (skip-whitespace string opportunity)))
+                           (setf best-break pos)
+                           (return-from breaks-rec nil)))
+                        ((null best-break)
+                         (return-from breaks-rec
+                           (if (> offset 0)
+                               (list* start (breaks-rec 0 margin start end i))
+                               (let* ((char-breaks (%line-breaks-1 string width
+                                                                   offset margin
+                                                                   start opportunity))
+                                      (new-start (alexandria:last-elt char-breaks)))
+                                 (append char-breaks (breaks-rec 0 margin new-start end i))))))
+                        (T #1=(return-from breaks-rec
```

Comment:
why is `T` upper case?

Reply:
I had this remark made twice today! what a coincidence. I have a habit of putting capital T (and for consistency NIL) where they have a special meaning because I find small-case t as a single letter distinctive and easy to miss. Do you want me to change that?

Reply:
I would say stick to what rest of the code base does (I try to do that by not using keywords in `loop`).

Reply:
fixed (force-pushed)